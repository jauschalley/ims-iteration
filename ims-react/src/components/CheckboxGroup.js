import React from 'react';
import PropTypes from 'prop-types';

const InputFeedback = ({error}) =>
  error ? <div className={"input-feedback"}>{error}</div> : null;

const Checkbox = ({
  field: {name, value, onChange, onBlur},
  form: {errors, touched, setFieldValue},
  id,
  label,
  className,
  ...props
}) => {
  return (
    <span class="margin-right-60">
      <input
        name={name}
        id={id}
        type="checkbox"
        value={value}
        checked={value}
        onChange={onChange}
        onBlur={onBlur}
        className="margin-right-5"
      />
      <label htmlFor={id} className="">{label}</label>
      {touched[name] && <InputFeedback error={errors[name]} />}
    </span>
  );
};

class CheckboxGroup extends React.Component {
  handleChange = event => {
    const target = event.currentTarget;
    let valueArray = [...this.props.value] || [];

    if (target.checked) {
      valueArray.push(target.id);
    } else {
      valueArray.splice(valueArray.indexOf(target.id), 1);
    }

    this.props.onChange(this.props.id, valueArray);
  };

  handleBlur = () => {
    this.props.onBlur(this.props.id, true);
  };

  render() {
    const {value, error, touched, label, className, children} = this.props;

    const classes = `input-field ${className}`;

    return (
      <div className={classes}>
        <fieldset>
          <legend>{label}</legend>
          {React.Children.map(children, child => {
            return React.cloneElement(child, {
              field: {
                value: value.includes(child.props.id),
                onChange: this.handleChange,
                onBlur: this.handleBlur
              }
            });
          })}
          {touched && <InputFeedback error={error} />}
        </fieldset>
      </div>
    );
  }
}

CheckboxGroup.propTypes = {
  value: PropTypes.string,
  id: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func
}

export {CheckboxGroup, Checkbox};