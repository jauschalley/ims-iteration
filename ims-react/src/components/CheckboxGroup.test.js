import React from 'react';
import renderer from 'react-test-renderer';
import {CheckboxGroup} from './CheckboxGroup';

describe('<CheckboxGroup />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <CheckboxGroup 
          value="blah"
          id="1"
          onChange={() => {}}
          onBlur={() => {}}
        />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});