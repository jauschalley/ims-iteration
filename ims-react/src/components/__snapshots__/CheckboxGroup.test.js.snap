// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxGroup /> renders correctly 1`] = `
<div
  className="input-field undefined"
>
  <fieldset>
    <legend />
  </fieldset>
</div>
`;
